corehq/apps/app_manager/templates/app_manager/module_view_report.html

```diff
@@ -1,24 +1,76 @@
 {% extends "app_manager/managed_app.html" %}
 {% load xforms_extras %}
 {% load hq_shared_tags %}
+{% load reports_core_tags %}
 {% load i18n %}
+
+{% block head %}{{ block.super }}
+    <link href="{% static 'hqwebapp/js/lib/select2/select2.css' %}" rel="stylesheet"/>
+{% endblock %}
 {% block js %}{{ block.super }}
     <script src="{% static 'style/ko/knockout_bindings.ko.js' %}"></script>
     <script src="{% static 'hqwebapp/js/lib/underscore-1.4.4.js' %}"></script>
     <script src="{% static 'app_manager/js/report-module.js' %}"></script>
+    <script src="{% static 'hqwebapp/js/lib/select2/select2.js' %}"></script>
 {% endblock %}
 {% block js-inline %}{{ block.super }}
     <script>
-        $(function () {
+{#        $(function () {#}
             var saveURL = "{% url "corehq.apps.app_manager.views.edit_report_module" domain app.id module.id %}";
-            ko.applyBindings(new ReportModule({
+            var reportModule = new ReportModule({
                 moduleName: {{ module.name|JSON }},
-                availableReports: {{ all_reports|JSON }},
-                currentReports: {{ current_reports|JSON }},
+                availableReports: {{ all_reports|JSON }}, // structure for all reports
+                currentReports: {{ current_reports|JSON }}, // config data for app reports
                 saveURL: saveURL,
                 lang: {{ lang|JSON }}
-            }), $("#report-module-edit")[0]);
-        });
+            });
+            ko.applyBindings(reportModule, $("#report-module-edit")[0]);
+
+
+        var select2s = $('.choice_filter');
+        for(var i = 0; i < select2s.length; i++) {
```

Comment:
how much of this was copy/pasted from [dynamic_choice_list.js](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports_core/templates/reports_core/filters/dynamic_choice_list_filter/dynamic_choice_list.js#L2) (and is there any way to share more code?)